src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -90,8 +93,47 @@ public DataWriter(HdfsSinkConnectorConfig connectorConfig, SinkTaskContext conte
       conf = new Configuration();
       conf.addResource(new Path(hadoopConfDir + "/core-site.xml"));
       conf.addResource(new Path(hadoopConfDir + "/hdfs-site.xml"));
+
+      String connectSiteConfDir = connectorConfig.getString(HdfsSinkConnectorConfig.CONNECT_SITE_CONF_DIR_CONFIG);
+      if (!connectSiteConfDir.equals("")) {
+        conf.addResource(new Path(connectSiteConfDir + "/connect-site.xml"));
+      }
       log.info(conf.toString());
 
+      if (conf.get("hadoop.security.authentication").equals("kerberos")) {
+        String principal = conf.get("connect.hdfs.principal");
+        String keytab = conf.get("connect.hdfs.keytab");
+
+        final UserGroupInformation ugi = UserGroupInformation.loginUserFromKeytabAndReturnUGI(principal,
+                                                                                              keytab);
+        log.info("Login as: " + ugi.getUserName());
+
+        final long renewPeriod = connectorConfig.getLong(HdfsSinkConnectorConfig.KERBEROS_TICKET_RENEW_PERIOD_MS_CONFIG);
+
+        isRunning = true;
+        ticketRenewThread = new Thread(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              synchronized (DataWriter.this) {
+                while (isRunning) {
+                  wait(renewPeriod);
+                  if (isRunning) {
+                    ugi.reloginFromKeytab();
+                  }
+                }
+              }
+            } catch (IOException e) {
+              log.error("Error renewing the ticket", e);
```

Comment:
Is catching this outside the loop going to cause problems if we, e.g., have a transient connectivity issue? I think it would kill this renewal thread.

I think the exception handling needs to happen in the `while` loop, and we probably need some smaller backoff for retrying in the case of failures (since you probably don't want to wait the entire renewal period if one attempt failed since you need to get it renewed).